src/search.cpp

```diff
@@ -346,35 +349,55 @@ void realMain(LambdaOptions const & options)
             "Searching and extending hits on-line...progress:\n"
             "0%  10%  20%  30%  40%  50%  60%  70%  80%  90%  100%\n|");
 
-    double start = sysTime();
-
+    double   start       = sysTime();
     uint64_t lastPercent = 0;
 
-    bio::io::seq::record r{.id   = std::string{},
-                           .seq  = std::vector<typename TGlobalHolder::TOrigQryAlph>{},
-                           .qual = std::ignore};
-    bio::io::seq::reader reader{options.queryFile, bio::io::seq::reader_options{.record = r}};
-
-    auto file_view = reader | views::async_input_buffer(globalHolder.records_per_batch * options.threads);
+    /* Setup lazy-loader for query (if set in options) */
+    decltype(createQryView(globalHolder, options)) file_view;
+    if (options.lazyQryFile)
+        file_view = createQryView(globalHolder, options);
 
     SEQAN_OMP_PRAGMA(parallel)
     {
         TLocalHolder localHolder(options, globalHolder);
 
+        /* Define region of query this thread get (eager-loading only) */
```

Comment:
gets

Reply:
Should this be in an if statement because it does not apply to lazy-loading?

Reply:
> Should this be in an if statement because it does not apply to lazy-loading?

The variables need to be defined because of scope either way. It only happens once when the thread starts, so it doesn't influence performance.